src/adapter/output/visit_repository_impl.py

```
@@ -24,18 +23,28 @@ def save(self, visit_log: VisitLogEntity) -> VisitLogEntity:
+                VisitLogEntity.visitor_ip == ip,
+                func.date(VisitLogEntity.visit_date) == func.current_date()
+            ))
+
+            return self.db.execute(stmt).scalar()
+        except SQLAlchemyError as e:
+            self.db.rollback()
+            logger.error(f"❌ Database error in exist_by_ip_and_date(): {e}") 
+            raise HTTPException(
+                status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                detail="Internal Server Error - Unable to check visit existence"
+            )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)